### PR TITLE
Update Dockerfile

### PR DESCRIPTION
**Pull request by amitkumarj441**
The updated version of the dockerfile reduces the size of the docker image which is created. This would drastically reduce the size of the docker image.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -7,16 +7,15 @@ RUN pacman -Syu --noconfirm nodejs npm unrar git python2 base-devel
...
-RUN git clone --branch $BCOIN_BRANCH $BCOIN_REPO /code/node_modules/bcoin
+RUN mkdir -p /code/node_modules/bcoin /data && \
+    git clone --branch $BCOIN_BRANCH $BCOIN_REPO /code/node_modules/bcoin
 
 # Installation
 WORKDIR /code/node_modules/bcoin
-RUN npm install --production
-
+RUN npm install --production && \
 # Cleanup
```
Comment (nodech) [inline Dockerfile:16]: This will skip the next lines

File: Dockerfile [hunk 2]
```diff
@@ -7,16 +7,15 @@ RUN pacman -Syu --noconfirm nodejs npm unrar git python2 base-devel
...
-RUN npm uninstall node-gyp
-RUN pacman -Rns --noconfirm unrar python2 && \
+    npm uninstall node-gyp
```
Comment (nodech) [inline Dockerfile:17]: lacking `&& \`

**Review by nodech — COMMENTED** [review]
I don't believe it will give improvements in size, most of the size comes from base image and packets

**Comment by BluSyn** [comment]
Included in #4

**Pull request closed without merging**